app/commands/git/sync_exercise.rb

```diff
@@ -0,0 +1,92 @@
+module Git
+  class SyncExercise
+    include Mandate
+    initialize_with :exercise
+
+    def call
+      lookup_head_and_current_commit
+      update_exercise! unless exercise_synced_to_head?
+    end
+
+    private
+    attr_reader :current_commit, :head_commit
+
+    def lookup_head_and_current_commit
+      exercise.git.update!
+
+      @current_commit = exercise.git.lookup_commit(exercise.synced_to_git_sha)
+      @head_commit = exercise.git.head_commit
+    end
+
+    def exercise_synced_to_head?
+      current_commit.oid == head_commit.oid
+    end
+
+    def update_exercise!
+      update_concept_exercise! if exercise.concept_exercise?
+      update_practice_exercise! if exercise.practice_exercise?
+      # # TODO: validate exercise to prevent invalid exercise data
+    end
+
+    def update_concept_exercise!
+      if exercise_modified?
+        exercise.update!(
+          slug: config_exercise[:slug],
+          title: config_exercise[:name],
+          git_sha: head_commit.oid,
+          synced_to_git_sha: head_commit.oid
+        )
+      else
+        exercise.update!(
+          synced_to_git_sha: head_commit.oid
+        )
+      end
+    end
+
+    def update_practice_exercise!
+      # TODO
+    end
+
+    def exercise_modified?
+      config_exercise_modified? || exercise_files_modified?
+    end
+
+    def config_exercise_modified?
+      return false unless track_config_modified?
+
+      config_exercise[:slug] != exercise.slug ||
+        config_exercise[:name] != exercise.title ||
+        config_exercise[:deprecated] != exercise.deprecated
+    end
+
+    def exercise_files_modified?
+      return false if current_commit.oid == head_commit.oid
+
+      diff.each_delta.any? do |delta|
+        delta.old_file[:path] == exercise.track.git.config_filepath ||
+          delta.new_file[:path] == exercise.track.git.config_filepath
```

Comment:
Is this the wrong filepath?

Reply:
No, because some of the track metadata is defined in the track's config file

Reply:
It is. Work in progress :)